app/src/Entity/Appointment.php

```diff
@@ -0,0 +1,187 @@
+<?php
+
+namespace App\Entity;
+
+use ApiPlatform\Metadata\ApiResource;
+use App\Repository\AppointmentRepository;
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\Mapping as ORM;
+
+#[ORM\Entity(repositoryClass: AppointmentRepository::class)]
+#[ApiResource]
+class Appointment
+{
+    #[ORM\Id]
+    #[ORM\GeneratedValue]
+    #[ORM\Column]
+    private int $id;
+
+    #[ORM\ManyToOne(inversedBy: 'appointments')]
+    private User $user_id;
+
+    #[ORM\Column(length: 255)]
+    private string $title;
+
+    #[ORM\Column(type: Types::TEXT, nullable: true)]
+    private ?string $description = null;
+
+    #[ORM\Column]
+    private \DateTimeImmutable $schedule;
+
+    #[ORM\Column(type: Types::DECIMAL, precision: 10, scale: '0')]
+    private string $price;
+
+    #[ORM\Column(length: 255)]
+    private string $status;
+
+    #[ORM\Column]
+    private ?\DateTimeImmutable $created_at = null;
+
+    #[ORM\Column(nullable: true)]
+    private ?\DateTimeImmutable $updated_at = null;
+
+    #[ORM\JoinTable(name: 'appointments_media_objects')]
+    #[ORM\JoinColumn(name: 'appointment_id', referencedColumnName: 'id')]
+    #[ORM\InverseJoinColumn(name: 'media_object_id', referencedColumnName: 'id', unique: true)]
+    #[ORM\ManyToMany(targetEntity: 'MediaObject')]
+    // #[ORM\OneToMany(mappedBy: 'appointment', targetEntity: MediaObject::class)]
+    private Collection $media_object_id;
+
+    public function __construct()
+    {
+        $this->media_object_id = new ArrayCollection();
+    }
```

Comment:
Change the constructor code to the following:

```
public function __construct(
    User $user,
    string $title,
    ?string $description,
    DateTime $schedule,
    float $price
) {
    $this->user = $user;
    $this->title = $title;
    $this->description = $description;
    $this->schedule = $schedule;
    $this->price = $price;
    $this->mediaObjects = new ArrayCollection();
    $this->createdAt = new DateTimeImmutable();
}
```

Please, apply these to all the other objects.

For more information of this approach, you can find it here:

https://youtu.be/WW2qPKukoZY?t=961